soc/CMakeLists.txt

```diff
@@ -9,8 +9,38 @@ if(_SOC_IS_IN_TREE)
 endif()
 unset(_SOC_IS_IN_TREE)
 
-if(EXISTS ${SOC_DIR}/${ARCH}/CMakeLists.txt)
-  add_subdirectory(${SOC_DIR}/${ARCH} soc/${ARCH})
-else()
-  add_subdirectory(${SOC_DIR}/${ARCH}/${SOC_PATH} soc/${ARCH}/${SOC_PATH})
+if(HWMv1)
+  if(EXISTS ${SOC_DIR}/${ARCH}/CMakeLists.txt)
+    add_subdirectory(${SOC_DIR}/${ARCH} soc/${ARCH})
+  else()
+    add_subdirectory(${SOC_DIR}/${ARCH}/${SOC_PATH} soc/${ARCH}/${SOC_PATH})
+  endif()
+elseif(HWMv2)
+  # Below is inclusion of HWMv2 SoC CMake lists.
+  string(TOUPPER SOC_FAMILY_${SOC_FAMILY}_DIR family_setting_dir)
+  string(TOUPPER SOC_SERIES_${SOC_SERIES}_DIR series_setting_dir)
+  string(TOUPPER SOC_${SOC_NAME}_DIR soc_setting_dir)
+
+  if(DEFINED ${soc_setting_dir})
+    add_subdirectory(${${soc_setting_dir}})
```

Comment:
Is it correctly understood that this will not support out-of-tree SoC's as the binary directory is not passed to add_subdirectory as it when using HWMv1?

Reply:
oot SoC's are definitely intended to be supported through the use of `SOC_ROOT`.

This part is an oversight, so thanks for spotting 👍 .

Reply:
fixed

Reply:
Thanks a lot. I took the latest patch and now oot SoCs works for me.